test/integration/targets/filesystem/tasks/overwrite_another_fs.yml

```diff
@@ -3,11 +3,20 @@
 
 - name: 'Create a vfat filesystem'
   command: 'mkfs.vfat {{ dev }}'
-  when: ansible_system != 'FreeBSD'
```

Comment:
Ah, right. I thought it just tests that it doesn't overwrite any existing filesystem (which is a necessary test too). I can change it to another Linux FS, but can't think of any common ones that the module doesn't handle.

Reply:
A swap partition (created using `mkswap`) could be used.

Reply:
`newfs` does not seem to work on image files, so I'll use `mkfs.ext2`.

Like you say it is different in that it now tests a filesystem supported by the module, but it still tests that an existing filesystem is not overwritten which I think is the important part. I don't think the control flow is any different (apart from the fact that `FILESYSTEMS.get` returns a class instead of `None`).

Reply:
Sorry didn't see your previous comment. I'll update it to use a swap partition instead.

Reply:
`mkswap` doesn't exist on FreeBSD (and `newfs` doesn't work). Any ideas what to use instead?

Reply:
> Ah, right. I thought it just tests that it doesn't overwrite any existing filesystem (which is a necessary test too).
This use case (supported filesystem aren't overwritten when force isn't used) is already tested [there](https://github.com/kwohlfahrt/ansible/blob/db7513ec2f9721333275e2deafa6c576de2fa526/test/integration/targets/filesystem/tasks/create_fs.yml#L19).

Reply:
Ah, I see. Should this test be skipped on FreeBSD then?

Reply:
ok